quickfixj-core/src/main/java/quickfix/Message.java

```diff
@@ -578,7 +578,13 @@ private void parseGroup(String msgType, StringField field, DataDictionary dd, Da
         final int[] fieldOrder = groupDataDictionary.getOrderedFields();
         int previousOffset = -1;
         final int groupCountTag = field.getField();
-        final int declaredGroupCount = Integer.parseInt(field.getValue());
+       // QFJ-533
+        int declaredGroupCount = 0;
+        try {
+          declaredGroupCount = Integer.parseInt(field.getValue());
+        } catch (final NumberFormatException e) {
+            throw new InvalidMessage("NumberFormatException: Repeating group count requires an Integer but found:" +field.getValue());
```

Comment:
Instead of making "NumberFormatException" part of the string, you should provide it as the cause.

Reply:
good point, I'll amend the PRs